lib/Controller/ConfigController.php

```
@@ -193,19 +196,33 @@ private function setIntegrationConfig(array $values): array {
 			}
 		}
 		$openProjectGroupFolderFileId = null;
+		$appPassword = null;
 
-		if (key_exists('setup_group_folder', $values) && $values['setup_group_folder']) {
+		if (key_exists('setup_group_folder', $values) && $values['setup_group_folder'] === true) {
 			$isSystemReady = $this->openprojectAPIService->isSystemReadyForGroupFolderSetUp();
 			if ($isSystemReady) {
 				$password = $this->secureRandom->generate(10, ISecureRandom::CHAR_HUMAN_READABLE);
 				$user = $this->userManager->createUser(Application::OPEN_PROJECT_ENTITIES_NAME, $password);
 				$group = $this->groupManager->createGroup(Application::OPEN_PROJECT_ENTITIES_NAME);
 				$group->addUser($user);
 				$this->subAdminManager->createSubAdmin($user, $group);
+				// finish the setup of the user by doing a PROPFIND request
+				// without this request we can get LockException or NotFoundException
+				// after creating the group folder
+				$this->finishUserSetup($password);
 				$openProjectGroupFolderFileId = $this->openprojectAPIService->createGroupfolder();
 			}
 		}
 
+//		 creates or replace the app password
+		if (key_exists('setup_app_password', $values) && $values['setup_app_password'] === true) {
+			$this->openprojectAPIService->deleteAppPassword();
+			if (!$this->userManager->userExists(Application::OPEN_PROJECT_ENTITIES_NAME)) {
+				throw new \Exception('User "' . Application::OPEN_PROJECT_ENTITIES_NAME . '" does not exists to create application password');
```

Should we throw the specific exception here and catch with and return it with `404`  here instead of the general exception  , I think this will return  `400` 
```
NotFoundException | NoUserException 
```